vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/main/java/com/vaadin/flow/component/spreadsheet/Spreadsheet.java

```diff
@@ -283,538 +245,429 @@ public void setWidth(String width) {
 
     private Set<String> invalidFormulaCells;
 
-    private boolean hasActions;
-
     private HashMap<String, OverlayInfo> overlays;
 
     private ArrayList<MergedRegion> mergedRegions;
 
-    private boolean displayGridlines = true;
-
-    private boolean displayRowColHeadings = true;
-
     private int verticalSplitPosition = 0;
     private int horizontalSplitPosition = 0;
 
     private String infoLabelValue;
 
     private boolean workbookChangeToggle;
 
-    private String invalidFormulaErrorMessage = "Invalid formula";
-
-    private boolean lockFormatColumns = true;
-
-    private boolean lockFormatRows = true;
-
-    private List<String> namedRanges;
-
     public String getDescription() {
         return description;
     }
 
-    public ContentMode getDescriptionContentMode() {
-        return descriptionContentMode;
-    }
-
-    public String getCaption() {
-        return caption;
-    }
-
-    public List<String> getStyles() {
-        return styles;
-    }
-
-    public String getPrimaryStyleName() {
-        return primaryStyleName;
-    }
-
-    public String getErrorMessage() {
-        return errorMessage;
-    }
-
-    public ErrorLevel getErrorLevel() {
-        return errorLevel;
-    }
-
-    public boolean isCaptionAsHtml() {
-        return captionAsHtml;
-    }
-
-    public Map<String, String> getResources() {
-        return resources;
-    }
-
-    public Set<String> getRegisteredEventListeners() {
-        return registeredEventListeners;
-    }
-
-    public int getColumnBufferSize() {
-        return columnBufferSize;
-    }
-
-    public int getCols() {
+    protected int getCols() {
         return cols;
     }
 
-    public List<GroupingData> getColGroupingData() {
+    private List<GroupingData> getColGroupingData() {
         return colGroupingData;
     }
 
-    public List<GroupingData> getRowGroupingData() {
-        return rowGroupingData;
-    }
-
-    public int getColGroupingMax() {
+    protected int getColGroupingMax() {
         return colGroupingMax;
     }
 
-    public int getRowGroupingMax() {
+    protected int getRowGroupingMax() {
         return rowGroupingMax;
     }
 
-    public boolean isColGroupingInversed() {
+    protected boolean isColGroupingInversed() {
         return colGroupingInversed;
     }
 
-    public boolean isRowGroupingInversed() {
+    protected boolean isRowGroupingInversed() {
         return rowGroupingInversed;
     }
 
-    public float getDefRowH() {
+    protected float getDefRowH() {
```

Comment:
> If removal isn't an option for some methods, rather make them package-private. No need to add this strange undocumented API available for users.

Why are these API still protected? Were they protected in V8 TabSheet?

Reply:
The protected ones are now used in other classes. For this example, it is used in SpreadsheetFactory.

Reply:
SpreadsheetFactory is in the same package as SpreadSheet, so it can be package-private, right? (to not expose this undocumented internal API for extensions)

Reply:
Sure, done.